test/instance-replacement-tests.js

```diff
@@ -24,6 +24,27 @@ test('Instance replacement', assert => {
   })();
 });
 
+test('Instance not replaced', assert => {
+  const newInstance = undefined;
+
+  compose({
+    initializers: [
+      () => {
+        return newInstance;
```

Comment:
I need to rename test to 
> 'Instance should not be replaced'

I return `undefined` to make sure it was not used as an object instance. This case was not covered.